### PR TITLE
Prevent panics when bad state is found - e.g. no meta.json or no indexing dir

### LINKED ISSUE #21: panic when indexing dir that does not exist
```
$ tantivy index --index /xxx
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "Indexing failed : Error(PathDoesNotExist(\"/xxx\"), State { next_error: None, backtrace: None })"', libcore/result.rs:945:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

### PR DESCRIPTION
**Pull request by ehiggs**
This fixes #21 and #22.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -137,5 +136,12 @@ fn main() {
         "bench" => run_bench_cli,
         _ => panic!("Subcommand {} is unknown", subcommand)
     };
-    run_cli(options).unwrap();
+
+    if let Err(ref e) = run_cli(options) {
+        use std::io::Write;
```
Comment (fulmicoton) [inline src/main.rs]: Could you move the `use` statement at the top of the file?

File: src/main.rs [hunk 2]
```diff
@@ -137,5 +136,12 @@ fn main() {
...
+        let stderr = &mut ::std::io::stderr();
+        let errmsg = "Error writing ot stderr";
+        writeln!(stderr, "{}", e).expect(errmsg);
+        ::std::process::exit(1);
```
Comment (fulmicoton) [inline src/main.rs]: ideally `use std::process` at the top of the file

File: src/errors.rs [hunk 3]
```diff
@@ -0,0 +1 @@
+error_chain!{}
```
Comment (fulmicoton) [inline src/errors.rs]: Is `error_chain!` useful here?

**Comment by fulmicoton** [comment]
Thanks! I put a bunch of comments.

**Comment by ehiggs** [comment]
Should be all fixed!

**Comment by fulmicoton** [comment]
Thanks!

**Pull request merged**